controllers/pytorch/pytorchjob_controller.go

```diff
@@ -230,7 +252,18 @@ func (r *PytorchJobReconciler) SetClusterSpec(ctx context.Context, job interface
 		rank++
 	}
 
-	totalReplicas := int(k8sutil.GetTotalReplicas(pytorchJob.Spec.PyTorchReplicaSpecs))
+	totalReplicas := int(k8sutil.GetTotalExcludedReplicas(pytorchJob.Spec.PyTorchReplicaSpecs, v1.JobReplicaTypeAIMaster))
+	enableElasticScaling := pytorchJob.Annotations[v1.AnnotationEnableElasticTraining] == "true"
+	if enableElasticScaling && !masterRole && rtype != "aimaster" {
+		AddImageWarmupForWorker(podTemplate, r.GetDefaultContainerName())
```

Comment:
Is this trying to warmup the worker while waiting for master to be ready, but kubelet will run initContainer sequentially.

Reply:
it's true that kubelet will run initContainers sequentially first, warmup container does nothing but pull image of pytorch container, then the next initContainer in order will wait for master to be ready.